collector/eventhub/src/main/java/zipkin/collector/eventhub/IEventProcessorHostFactory.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.eventhub;
+
+public interface IEventProcessorHostFactory {
```

Comment:
in java, these sorts of interfaces paint us into corners, particularly as adding parameters or methods to an interface is api breaking. In general, we like to keep api compatibility as long as possible, and one way to to not expose things that would make it possible to break api. Here's an example explanation, but there are similar ones in the book "effective java" and many codebases including zipkin and guava.
https://github.com/openzipkin/brave/blob/master/HACKING.md#hacking-brave

What's the use case for changing this? Maybe we can figure out a way to avoid exposing more public types.